x/vesting/keeper/grpcs_query.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/tharsis/evmos/x/vesting/types"
+)
+
+var _ types.QueryServer = Keeper{}
+
+// Balances returns the locked, unvested and vested amount of tokens for a
+// clawback vesting account
+func (k Keeper) Balances(
+	goCtx context.Context,
+	req *types.QueryBalancesRequest,
+) (*types.QueryBalancesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	addr, err := sdk.AccAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Get vesting account
+	acc := k.accountKeeper.GetAccount(ctx, addr)
+	if acc == nil {
+		return nil, status.Errorf(codes.NotFound,
+			"account for address '%s'", req.Address,
+		)
+	}
+
+	// Check if clawback vesting account
+	clawbackAccount, isClawback := acc.(*types.ClawbackVestingAccount)
```

Comment:
Is it possible to use the interface so that this query supports other vesting types? Otherwise, we can leave it as is